### PR TITLE
Avoid using a guava classes which is not compatible with later versions of guava.

### PR DESCRIPTION
**Pull request by anwar6953**
For instance, in Integration test framework, avoid using a StopWatch class which is not compatible with later versions of guava (18, for instance).

### CODE REVIEW

**Review by CuriousVini — APPROVED** [review]
lgtm

File: cdap-unit-test/src/test/java/co/cask/cdap/service/FileUploadServiceTestRun.java [hunk 1]
```diff
@@ -95,7 +95,7 @@ public void testFileUploadService() throws Exception {
       // There should be one file under the partition directory
       List<Location> locations = partition.getLocation().list();
       Assert.assertEquals(1, locations.size());
-      Assert.assertArrayEquals(content, ByteStreams.toByteArray(Locations.newInputSupplier(locations.get(0))));
+      Assert.assertArrayEquals(content, ByteStreams.toByteArray(locations.get(0)::getInputStream));
```
Comment (chtyim) [inline cdap-unit-test/src/test/java/co/cask/cdap/service/FileUploadServiceTestRun.java]: you do need to close the input stream.

File: cdap-unit-test/src/test/java/co/cask/cdap/spark/SparkTest.java [hunk 2]
```diff
@@ -277,7 +277,7 @@ public boolean apply(Location input) {
       });
 
       // Verify the result
-      List<String> lines = CharStreams.readLines(CharStreams.newReaderSupplier(Locations.newInputSupplier(outputFile),
+      List<String> lines = CharStreams.readLines(CharStreams.newReaderSupplier(outputFile::getInputStream,
```
Comment (chtyim) [inline cdap-unit-test/src/test/java/co/cask/cdap/spark/SparkTest.java]: Same here. Need to close the stream.

File: cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java [hunk 3]
```diff
@@ -416,7 +416,7 @@ private void copyJarFile(URL jarURL, File file) {
     try {
       JarURLConnection jarConn = (JarURLConnection) jarURL.openConnection();
       try {
-        Files.copy(Resources.newInputStreamSupplier(jarConn.getJarFileURL()), file);
+        com.google.common.io.Files.copy(Resources.newInputStreamSupplier(jarConn.getJarFileURL()), file);
```
Comment (chtyim) [inline cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java]: It is better just open the stream

```java
  try (InputStream is = jarConn.getJarFileURL().openStream()) {
    Files.copy(is, file.toPath());
  }
```

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/router/NettyRouterPipelineTest.java [hunk 4]
```diff
@@ -221,7 +221,7 @@ private void deploy(int num) throws Exception {
       urlConn.setDoOutput(true);
       urlConn.setDoInput(true);
 
-      ByteStreams.copy(Locations.newInputSupplier(programJar), urlConn.getOutputStream());
+      ByteStreams.copy(programJar::getInputStream, urlConn.getOutputStream());
```
Comment (chtyim) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/router/NettyRouterPipelineTest.java]: Need to close the stream

**Comment by anwar6953** [comment]
I believe that the method I’m calling will close it internally after
calling the getInputStream method.

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/router/NettyRouterPipelineTest.java (hunk 4 above)
Comment (anwar6953) [reply cdap-gateway/src/test/java/co/cask/cdap/gateway/router/NettyRouterPipelineTest.java]: I believe that the method I’m calling will close it internally after
it calls the `getInputStream` method.

File: cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java (hunk 3 above)
Comment (anwar6953) [reply cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java]: The current code also closes the `jarConn.getJarFile().close()`.
Do we still need to do that?

File: cdap-common/src/main/java/co/cask/cdap/common/internal/guava/ClassPath.java [hunk 5]
```diff
@@ -136,7 +136,10 @@ public static ClassPath from(URI uri, ClassLoader classLoader) throws IOExceptio
    * @since 16.0
    */
   public ImmutableSet<ClassInfo> getAllClasses() {
-    return FluentIterable.from(resources).filter(ClassInfo.class).toImmutableSet();
+    return resources.stream()
+      .filter(ClassInfo.class::isInstance)
+      .map(resourceInfo -> (ClassInfo) resourceInfo)
```
Comment (chtyim) [inline cdap-common/src/main/java/co/cask/cdap/common/internal/guava/ClassPath.java]: use `.map(ClassInfo.class::cast)` instead.

File: cdap-common/src/main/java/co/cask/cdap/common/internal/guava/ClassPath.java [hunk 6]
```diff
@@ -136,7 +136,10 @@ public static ClassPath from(URI uri, ClassLoader classLoader) throws IOExceptio
...
+      .collect(Collectors.collectingAndThen(Collectors.toSet(), ImmutableSet::copyOf));
```
Comment (chtyim) [inline cdap-common/src/main/java/co/cask/cdap/common/internal/guava/ClassPath.java]: Use `Collectors.toCollection(LinkedHashSet::new)` instead fo preserve the behavior, which an `ImmutableSet` has iteration order to same as insertion order.

File: cdap-common/src/main/java/co/cask/cdap/common/internal/guava/ClassPath.java [hunk 7]
```diff
@@ -136,7 +136,10 @@ public static ClassPath from(URI uri, ClassLoader classLoader) throws IOExceptio
    * @since 16.0
    */
   public ImmutableSet<ClassInfo> getAllClasses() {
-    return FluentIterable.from(resources).filter(ClassInfo.class).toImmutableSet();
+    return resources.stream()
+      .filter(ClassInfo.class::isInstance)
+      .map(ClassInfo.class::cast)
+      .collect(Collectors.collectingAndThen(Collectors.toSet(), ImmutableSet::copyOf));
```
Comment (chtyim) [inline cdap-common/src/main/java/co/cask/cdap/common/internal/guava/ClassPath.java]: Use `Collectors.toCollection(LinkedHashSet::new)` instead to preserve the old behavior.

Comment (anwar6953) [reply cdap-common/src/main/java/co/cask/cdap/common/internal/guava/ClassPath.java]: Do you mean like this:
```java
.collect(Collectors.collectingAndThen(Collectors.toCollection(LinkedHashSet::new), ImmutableSet::copyOf));
```

**Comment by anwar6953** [comment]
https://builds.cask.co/browse/CDAP-DUT6431-1

**Comment by chtyim** [comment]
LGTM

**Comment by anwar6953** [comment]
Build is green; squashing and merging.

**Pull request merged**